numpy/_core/numeric.py

```diff
@@ -2505,7 +2525,7 @@ def array_equiv(a1, a2):
     except Exception:
         return False
 
-    return bool(asarray(a1 == a2).all())
+    return bool((a1 == a2).all())
```

Comment:
Nice! this does seem OK to remove, because we guarantee that `a == b` returns a NumPy boolean array/scalar.  So we don't run into problems even for object dtype!

Reply:
I can also send that as it's own PR if you wish.